pkg/controller/kibana/stackmon/metricbeat.tpl.yml

```diff
@@ -9,8 +9,8 @@ metricbeat.modules:
     hosts: ["{{ .URL }}"]
     username: {{ .Username }}
     password: {{ .Password }}
-    {{- if .IsSSL }}
-    ssl.certificate_authorities: ["{{ .SSLPath }}"]
+    {{- if .IsCA }}
+    ssl.certificate_authorities: ["{{ .CAPath }}"]
     ssl.verification_mode: "{{ .SSLMode }}"
```

Comment:
`verification_mode` is now predicated on the presence of a CA certificate. Will this not be a problem if users are using their custom certificates without a CA and those certificates are not issued for localhost?

Reply:
It is set to `certificate` which IIUC should verify that the provided certificate is signed by a trusted authority (CA), but does not perform any hostname verification ([doc](https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-ssl.html#ssl-common-config)). But you are right, this is a problem as reported by Michael [here](https://github.com/elastic/cloud-on-k8s/pull/5310#pullrequestreview-868801747). I'm a bit confused.

Reply:
Yes, you are right, it also needs to be set with a custom cert without a CA.

Reply:
I opted to keep `IsSSL`, add `ssl.enabled: {{ .IsSSL }}` and always set `ssl.verification_mode: certificate`.